pennylane_qiskit/qiskit_device2.py

```diff
@@ -250,11 +250,7 @@ def __init__(
         self._service = getattr(backend, "_service", None)
         self._session = session
 
-        # initial kwargs are saved and referenced every time the kwargs used for transpilation and execution
-        self._init_kwargs = kwargs
-        # _kwargs are used instead of the Options for performing raw sample based measurements (using old Qiskit API)
-        # the _kwargs are a combination of information from Options and _init_kwargs
-        self._kwargs = kwargs
+        kwargs["shots"] = shots
```

Comment:
What happens if the `options` have `shots`? They won't be able to override it directly? 🤔

Reply:
If you pass in options={"shots": 30}, you get a warning but options will override it. 

```        
  if "options" in kwargs:
      for key, val in kwargs.pop("options").items():
          if key in kwargs:
              warnings.warn(
                  "An overlap between what was passed in via options and what was passed in via kwargs was found."
                  f"The value set in options {key}={val} will be used."
              )
          kwargs[key] = val
     ```